### PR TITLE
Revert 40562 add validation for iam policy document sid

### LINKED ISSUE #40639: [Bug]: SID validation at plan time is invalidly applied to ALL policies
### Terraform Core Version

1.7

### AWS Provider Version

5.82

### Affected Resource(s)

aws_iam_policy_document

### Expected Behavior

We should not blanket apply plan time validation against the SID because each service may have it's own policies on allowed characters. This is a breaking change.

### Actual Behavior

We blanket apply the plan time validation against the SID.

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```terraform
resource "aws_kms_key" "name" {
  description         = "something"
  policy              = data.aws_iam_policy_document.name.json
}


data "aws_iam_policy_document" "name" {
  version                 = "2012-10-17"

  statement {
    sid    = "Allow spaces"
    effect = "Allow"
    actions = [
      "kms:Decrypt"
    ]
    resources = ["*"]
  }
}
```

### Steps to Reproduce

Create an IAM policy document resource, have the SID include spaces

Attach the policy as a KMS policy

### Debug Output

│   with aws_iam_policy_document.name,
│   on file line X, in data "aws_iam_policy_document" "name":
│   line#:     sid    = "Allow spaces"

### Panic Output

_No response_

### Important Factoids

_No response_

### References

https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md

https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-overview.html

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by DerekTBrown**
This reverts https://github.com/hashicorp/terraform-provider-aws/pull/40562, which is causing significant issues.

Closes #40639.

### CODE REVIEW

File: .changelog/40639.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+data-source/aws_iam_policy_document: Reverts 40652 due to issues. See: 40639.
```
Comment (YakDriver) [inline .changelog/40639.txt:2]:
```suggestion
data-source/aws_iam_policy_document: Reverts plan-time validation for `statement` `sid`
```

**Comment by YakDriver** [comment]
@DerekTBrown Thanks for quickly working on this! Please make the change and we'll get this merged.

**Comment by DerekTBrown** [comment]
> Please make the change and we'll get this merged.

Is there something I need to tweak about this PR?
Never made a change in this repo before.

**Comment by YakDriver** [comment]
Just commit the change in my review comment and we should be good.

**Review by YakDriver — CHANGES_REQUESTED** [review]

**Review by YakDriver — DISMISSED** [review]
Thank you for your contribution!

```console
% make t T=TestAccIAMPolicyDocumentDataSource_ K=iam
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/iam/... -v -count 1 -parallel 20 -run='TestAccIAMPolicyDocumentDataSource_'  -timeout 360m
2024/12/19 14:34:30 Initializing Terraform AWS Provider...
=== RUN   TestAccIAMPolicyDocumentDataSource_basic
=== PAUSE TestAccIAMPolicyDocumentDataSource_basic
=== RUN   TestAccIAMPolicyDocumentDataSource_singleConditionValue
=== PAUSE TestAccIAMPolicyDocumentDataSource_singleConditionValue
=== RUN   TestAccIAMPolicyDocumentDataSource_multipleConditionKeys
=== PAUSE TestAccIAMPolicyDocumentDataSource_multipleConditionKeys
=== RUN   TestAccIAMPolicyDocumentDataSource_duplicateConditionKeys
=== PAUSE TestAccIAMPolicyDocumentDataSource_duplicateConditionKeys
=== RUN   TestAccIAMPolicyDocumentDataSource_conditionWithBoolValue
=== PAUSE TestAccIAMPolicyDocumentDataSource_conditionWithBoolValue
=== RUN   TestAccIAMPolicyDocumentDataSource_source
=== PAUSE TestAccIAMPolicyDocumentDataSource_source
=== RUN   TestAccIAMPolicyDocumentDataSource_sourceList
=== PAUSE TestAccIAMPolicyDocumentDataSource_sourceList
=== RUN   TestAccIAMPolicyDocumentDataSource_sourceConflicting
=== PAUSE TestAccIAMPolicyDocumentDataSource_sourceConflicting
=== RUN   TestAccIAMPolicyDocumentDataSource_sourceListConflicting
=== PAUSE TestAccIAMPolicyDocumentDataSource_sourceListConflicting
=== RUN   TestAccIAMPolicyDocumentDataSource_override
=== PAUSE TestAccIAMPolicyDocumentDataSource_override
=== RUN   TestAccIAMPolicyDocumentDataSource_overrideList
=== PAUSE TestAccIAMPolicyDocumentDataSource_overrideList
=== RUN   TestAccIAMPolicyDocumentDataSource_noStatementMerge
=== PAUSE TestAccIAMPolicyDocumentDataSource_noStatementMerge
=== RUN   TestAccIAMPolicyDocumentDataSource_noStatementOverride
=== PAUSE TestAccIAMPolicyDocumentDataSource_noStatementOverride
=== RUN   TestAccIAMPolicyDocumentDataSource_duplicateSid
=== PAUSE TestAccIAMPolicyDocumentDataSource_duplicateSid
=== RUN   TestAccIAMPolicyDocumentDataSource_sourcePolicyValidJSON
=== PAUSE TestAccIAMPolicyDocumentDataSource_sourcePolicyValidJSON
=== RUN   TestAccIAMPolicyDocumentDataSource_overridePolicyDocumentValidJSON
=== PAUSE TestAccIAMPolicyDocumentDataSource_overridePolicyDocumentValidJSON
=== RUN   TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_stringAndSlice
=== PAUSE TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_stringAndSlice
=== RUN   TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_multiplePrincipals
=== PAUSE TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_multiplePrincipals
=== RUN   TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_multiplePrincipalsGov
=== PAUSE TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_multiplePrincipalsGov
=== RUN   TestAccIAMPolicyDocumentDataSource_version20081017
=== PAUSE TestAccIAMPolicyDocumentDataSource_version20081017
=== CONT  TestAccIAMPolicyDocumentDataSource_basic
=== CONT  TestAccIAMPolicyDocumentDataSource_overrideList
=== CONT  TestAccIAMPolicyDocumentDataSource_overridePolicyDocumentValidJSON
=== CONT  TestAccIAMPolicyDocumentDataSource_sourcePolicyValidJSON
=== CONT  TestAccIAMPolicyDocumentDataSource_duplicateConditionKeys
=== CONT  TestAccIAMPolicyDocumentDataSource_noStatementOverride
=== CONT  TestAccIAMPolicyDocumentDataSource_conditionWithBoolValue
=== CONT  TestAccIAMPolicyDocumentDataSource_version20081017
=== CONT  TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_multiplePrincipals
=== CONT  TestAccIAMPolicyDocumentDataSource_source
=== CONT  TestAccIAMPolicyDocumentDataSource_override
=== CONT  TestAccIAMPolicyDocumentDataSource_sourceListConflicting
=== CONT  TestAccIAMPolicyDocumentDataSource_sourceConflicting
=== CONT  TestAccIAMPolicyDocumentDataSource_sourceList
=== CONT  TestAccIAMPolicyDocumentDataSource_duplicateSid
=== CONT  TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_stringAndSlice
=== CONT  TestAccIAMPolicyDocumentDataSource_noStatementMerge
=== CONT  TestAccIAMPolicyDocumentDataSource_multipleConditionKeys
=== CONT  TestAccIAMPolicyDocumentDataSource_singleConditionValue
=== CONT  TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_multiplePrincipalsGov
    policy_document_data_source_test.go:394: skipping tests; current partition (aws) does not equal aws-us-gov
--- SKIP: TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_multiplePrincipalsGov (0.46s)
--- PASS: TestAccIAMPolicyDocumentDataSource_sourceListConflicting (3.00s)
--- PASS: TestAccIAMPolicyDocumentDataSource_overrideList (17.22s)
--- PASS: TestAccIAMPolicyDocumentDataSource_conditionWithBoolValue (17.24s)
--- PASS: TestAccIAMPolicyDocumentDataSource_duplicateConditionKeys (17.28s)
--- PASS: TestAccIAMPolicyDocumentDataSource_singleConditionValue (17.29s)
--- PASS: TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_stringAndSlice (17.29s)
--- PASS: TestAccIAMPolicyDocumentDataSource_sourceList (17.32s)
--- PASS: TestAccIAMPolicyDocumentDataSource_override (17.35s)
--- PASS: TestAccIAMPolicyDocumentDataSource_noStatementOverride (17.37s)
--- PASS: TestAccIAMPolicyDocumentDataSource_basic (17.37s)
--- PASS: TestAccIAMPolicyDocumentDataSource_sourcePolicyValidJSON (17.39s)
--- PASS: TestAccIAMPolicyDocumentDataSource_duplicateSid (17.42s)
--- PASS: TestAccIAMPolicyDocumentDataSource_noStatementMerge (17.43s)
--- PASS: TestAccIAMPolicyDocumentDataSource_StatementPrincipalIdentifiers_multiplePrincipals (17.43s)
--- PASS: TestAccIAMPolicyDocumentDataSource_sourceConflicting (17.43s)
--- PASS: TestAccIAMPolicyDocumentDataSource_multipleConditionKeys (17.43s)
--- PASS: TestAccIAMPolicyDocumentDataSource_version20081017 (21.04s)
--- PASS: TestAccIAMPolicyDocumentDataSource_overridePolicyDocumentValidJSON (22.04s)
--- PASS: TestAccIAMPolicyDocumentDataSource_source (22.56s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/iam	26.668s
```

**Review by ewbankkit — CHANGES_REQUESTED** [review]
Can we add an acceptance test that would have failed with the original change?

**Review by YakDriver — APPROVED** [review]
```console
% make t T=TestAccIAMPolicyDocumentDataSource_invalidSIDValid K=iam
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/iam/... -v -count 1 -parallel 20 -run='TestAccIAMPolicyDocumentDataSource_invalidSIDValid'  -timeout 360m
2024/12/19 14:58:24 Initializing Terraform AWS Provider...
=== RUN   TestAccIAMPolicyDocumentDataSource_invalidSIDValid
=== PAUSE TestAccIAMPolicyDocumentDataSource_invalidSIDValid
=== CONT  TestAccIAMPolicyDocumentDataSource_invalidSIDValid
--- PASS: TestAccIAMPolicyDocumentDataSource_invalidSIDValid (4.36s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/iam	8.546s
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Comment by DerekTBrown** [comment]
@ewbankkit @YakDriver Is this being tagged as a release?

**Pull request merged**